packages/augur-state/src/api/Markets.ts

```diff
@@ -1,33 +1,247 @@
-import {SortLimit} from './types';
+import { BigNumber } from "bignumber.js";
+import * as _ from "lodash";
+import { numTicksToTickSize } from "@augurproject/api";
+import { DB } from "../db/DB";
+import { MarketType, SortLimit } from "./types";
 
 export interface GetMarketsParamsSpecific {
-  universe: string,
-  creator?: string,
-  category?: string,
-  search?: string,
-  reportingState?: string,
-  disputeWindow?: string,
-  designatedReporter?: string,
-  maxFee?: number,
-  hasOrders?: boolean,
+  universe: string;
+  creator?: string;
+  category?: string;
+  search?: string;
+  reportingState?: string;
+  disputeWindow?: string;
+  designatedReporter?: string;
+  maxFee?: number;
+  hasOrders?: boolean;
 }
 
 export interface GetMarketsParams extends GetMarketsParamsSpecific, SortLimit {
 }
 
-export interface MarketsInfoParams {
-  marketIds: Array<string>
+export interface GetMarketsInfoParams {
+  marketIds: Array<string>;
 }
 
-export class Markets {
-  constructor() {
+export interface MarketInfoOutcome {
+  id: number;
+  price: string;
+  description: string|null;
+}
+
+export enum MarketInfoReportingState {
+  PRE_REPORTING = "PRE_REPORTING",
+  DESIGNATED_REPORTING = "DESIGNATED_REPORTING",
+  OPEN_REPORTING = "OPEN_REPORTING",
+  CROWDSOURCING_DISPUTE = "CROWDSOURCING_DISPUTE",
+  AWAITING_NEXT_WINDOW = "AWAITING_NEXT_WINDOW",
+  AWAITING_FINALIZATION = "AWAITING_FINALIZATION",
+  FINALIZED = "FINALIZED",
+  FORKING = "FORKING",
+  AWAITING_NO_REPORT_MIGRATION = "AWAITING_NO_REPORT_MIGRATION",
+  AWAITING_FORK_MIGRATION = "AWAITING_FORK_MIGRATION"
+}
+
+export interface MarketInfo {
+  id: string;
+  universe: string;
+  marketType: MarketType;
+  numOutcomes: number;
+  minPrice: string;
+  maxPrice: string;
+  cumulativeScale: string;
+  author: string;
+  creationTime: number;
+  creationBlock: number;
+  category:string;
+  volume: string;
+  openInterest: string;
+  reportingState: string;
+  needsMigration: boolean;
+  endTime: number;
+  finalizationBlockNumber: number|null;
+  finalizationTime: number|null;
+  description: string;
+  scalarDenomination: string|null;
+  details: string|null;
+  resolutionSource: string|null;
+  numTicks: string;
+  tickSize: string;
+  consensus: null, // TODO Update once consensus has been added to MarketFinalized Event
+  outcomes: Array<MarketInfoOutcome>;
+}
+
+export class Markets<TBigNumber> {
+  private readonly db: DB<TBigNumber>;
+
+  constructor(db: DB<TBigNumber>) {
+    this.db = db;
   }
 
   public async getMarkets(params: GetMarketsParams): Promise<void> {
     // TODO
   }
 
-  public async getMarketsInfo(params: MarketsInfoParams): Promise<void> {
-    // TODO
+  // TODO Add decorator code
+  // TODO Improve performance by using Promise.all for DB queries
+  public async getMarketsInfo(params: GetMarketsInfoParams): Promise<Array<MarketInfo>> {
```

Comment:
I recommend cutting this up into smaller functions that you can then unit test. You've already organized this function into paragraphs; I think most of those paragraphs could be their own function.

Reply:
Agreed, I was thinking the same thing. I was just planning to do that when I updated the getter to make use of the new log info.